### PR TITLE
feat(elixir): add Elixir examples

### PR DESCRIPTION
**Pull request by savonarola**

### CODE REVIEW

File: mqtt-client-Elixir/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Elixir MQTT client examples
+In this example we provide example code for [Elixir](https://elixir-lang.org/) to subscribe to topics and send messages.
+We use [`emqtt`](https://github.com/emqx/emqtt) client. For more documentation, see [`emqtt` documentation](https://github.com/emqx/emqtt/README.md).
```
Comment (thalesmg) [inline mqtt-client-Elixir/README.md:3]: The link to the readme is not working for me:

```suggestion
We use [`emqtt`](https://github.com/emqx/emqtt) client. For more documentation, see [`emqtt` documentation](https://github.com/emqx/emqtt/blob/master/README.md).
```

File: mqtt-client-Elixir/mqtt_tls.exs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  host: 'broker.emqx.io',
+  port: 8883,
+  client_id: "elixir_subscriber",
+  username: "user",
+  password: "password",
+  ssl: true,
+  ssl_opts: [
+    verify: :verify_peer,
+    server_name_indication: 'broker.emqx.io',
+    cacertfile: CAStore.file_path()
```
Comment (thalesmg) [inline mqtt-client-Elixir/mqtt_tls.exs:23]: Q: is this just an example?  Or were this module supposed to exist?  I couldn't find it.  :see_no_evil:

**Review by thalesmg — APPROVED** [review]

File: mqtt-client-Elixir/mqtt_tls.exs (hunk 2 above)
Comment (savonarola) [reply mqtt-client-Elixir/mqtt_tls.exs:23]: This module belongs to `castore` hex, insalled in the first line

Comment (thalesmg) [reply mqtt-client-Elixir/mqtt_tls.exs:23]: Oops, glossed over that line, sorry :see_no_evil:

File: mqtt-client-Elixir/README_ZH.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# Elixir MQTT 客户端示例
+在此示例中，我们提供了 [Elixir](https://elixir-lang.org/) 订阅主题和发送消息的示例代码。
+我们使用 [`emqtt`](https://github.com/emqx/emqtt) 客户端。 有关更多文档，请参阅 [`emqtt` 文档](https://github.com/emqx/emqtt/blob/master/README.md)。
+
+我们演示
+* 具有不同 QOS 的基本发布/订阅操作；
+* 通过 TLS 安全连接到代理的选项。
+
+＃＃ 跑步
```
Comment (zmstone) [inline mqtt-client-Elixir/README_ZH.md:9]:
```suggestion
## 运行
```

File: mqtt-client-Elixir/mqtt.exs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  username: "user",
+  password: "password"
+)
+{:ok, _} = :emqtt.connect(subscriber)
+
+qos1_granted = 1
+{:ok, %{} = _props, [^qos1_granted]} = :emqtt.subscribe(subscriber, topic, :qos1)
+
+{:ok, publisher} = :emqtt.start_link(
+  ## Note single quotes
```
Comment (zmstone) [inline mqtt-client-Elixir/mqtt.exs:25]: emqtt should really support binary string as hostnames.

**Comment by zmstone** [comment]
thanks Ilya.
just two minors.

File: mqtt-client-Elixir/mqtt.exs (hunk 4 above)
Comment (savonarola) [reply mqtt-client-Elixir/mqtt.exs:25]: https://github.com/emqx/emqtt/pull/187

File: mqtt-client-Elixir/mqtt.exs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  username: "user",
+  password: "password"
+)
+{:ok, _} = :emqtt.connect(subscriber)
+
+qos1_granted = 1
+{:ok, %{} = _props, [^qos1_granted]} = :emqtt.subscribe(subscriber, topic, :qos1)
+
+{:ok, publisher} = :emqtt.start_link(
+  host: 'broker.emqx.io',
```
Comment (thalesmg) [inline mqtt-client-Elixir/mqtt.exs:24]: Nit: maybe also use string (binary) here, now that emqtt supports it?

**Review by thalesmg — APPROVED** [review]

**Comment by savonarola** [comment]
@zmstone should we merge?

**Pull request merged**